src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c

```diff
@@ -2102,10 +2106,11 @@ bool wm_vuldet_check_generic_package(char * agent_id, sqlite3_stmt *pStmt, sqlit
             if (vulnerable == true) {
                 os_calloc(1, sizeof(cve_vuln_pkg), newPkg);
                 os_calloc(1, sizeof(cve_vuln_cond_NVD), newPkg->nvd_cond);
-                newPkg->version = pkg_version;
-                newPkg->bin_name = pkg_name;
-                newPkg->src_name = pkg_source;
-                newPkg->arch = pkg_arch;
+
```

Comment:
@JcabreraC @0xN3utr0n , this pointer assignment was causing "double free" issues when the list of packages was destroyed at the end of the vulnerability checking process. The reason is that this is within a while loop and the pointers could be assigned to multiples packages.